tests/xfail/xfail.go

```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package xfail
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// Enable replaces the default Fail handler with a XFail for one occurrence.
+// XFail skips the test once it fails and leaves a XFAIL label on the message.
+// It is useful to mark tests as XFail when one wants them to run even though they fail,
+// monitoring and collecting information.
+func Enable(reason string) {
+	RegisterFailHandler(func(m string, offset ...int) {
```

Comment:
Maybe worth that `RegisterFailHandler` will get the function as a parameter,
and the function that it gets will have `...OneShot` or something in its name, so it will be visible its one timer?

Reply:
Passing in such an argument will complicate the caller without any good reason. The operation is localized and should be trivial (expecting to enable xfail in a particular test and not globally).

For the same reason I do not think adding to the name "Once" is needed.